packages/runtime/test-runtime-utils/src/mocks.ts

```diff
@@ -534,6 +534,13 @@ export class MockFluidDataStoreRuntime extends EventEmitter
     public reSubmit(content: any, localOpMetadata: unknown) {
         return;
     }
+
+    /**
+     * Note: Search blob concept
+     */
+    public registerExtraSnapshotContract(contract: ISnapshotContract): void {
```

Comment:
Can you please glance at my hacking in Office-bohemia repo around usage pattern?
https://office.visualstudio.com/OC/_git/office-bohemia/pullrequest/582097?_a=files

Pasting relevant pieces here for easier consumption:
From usage POV, here is what looks like the most convinient:

```
  protected async hasInitialized() {
    ...
    this.addSummaryHandler(() => {
      return { path: ['search', '01'], this.getSearchContent() };
    });
  }
```
Exposing ITreeEntry or equivalence (summary tree APIs) is Ok, but it forces consumer to understand and take dependency on internal representation that is not interesting outside of fluid runtime. By hiding it we can make API simpler, but also make it possible to change things in the future without having back compat hit.

Reply:
Agreed re: exposing the tree API. Reading through your hack now